packages/token-bridge-sdk/src/hooks/useArbTokenBridge.ts

```diff
@@ -57,12 +64,36 @@ class TokenDisabledError extends Error {
     this.name = 'TokenDisabledError'
   }
 }
+
 export const useArbTokenBridge = (
   bridge: Bridge,
-  autoLoadCache = true
+  autoLoadCache = true,
+  l1Signer: Signer,
+  l2Signer: Signer
 ): ArbTokenBridge => {
   const [walletAddress, setWalletAddress] = useState('')
 
+  const [l1Network, setL1Network] = useState<L1Network>()
+  const [l2Network, setL2Network] = useState<L2Network>()
+
+  const [ethBridger, setEthBridger] = useState<EthBridger>()
+  const [erc20Bridger, setErc20Bridger] = useState<Erc20Bridger>()
+
+  useEffect(() => {
+    const sync = async function () {
+      const _l1Network = await getL1Network(l1Signer)
+      const _l2Network = await getL2Network(l2Signer)
+
+      setL1Network(_l1Network)
+      setL2Network(_l2Network)
+
+      setEthBridger(new EthBridger(_l2Network))
```

Comment:
might make sense to just instantiate the EthBridge or TokenBridge live in when deposit/withdraw is called? Just so there's less weird intermediate states to worry about (esp when switching networks, etc.). Not sure tho

Reply:
Agreed, will make it impossible to have them be `undefined`. :+1: